packages/react-sdk/src/index.tsx

```diff
@@ -30,108 +70,327 @@ const Context = createContext<BucketContext>({
+ * const isEnabled = useFlagIsEnabled('huddle');
+ * // true / false
  * ```
+ */
+export function useFlagIsEnabled(flagKey: BucketFlags) {
+  const { flags } = useContext<BucketContext>(Context).flags;
+  return flags[flagKey] ?? false;
+}
+
+/**
+ * Returns the state of a given feature flag for the current context, e.g.
  *
- * See the [Tracking SDK documentation](https://github.com/bucketco/bucket-tracking-sdk/blob/main/packages/tracking-sdk/README.md) for usage information.
+ * ```ts
+ * const huddleFlag = useFlag("huddle");
+ * // {
+ * //   "isLoading": false,
+ * //   "isEnabled": true,
+ * // }
+ * ```
  */
-export function useBucket() {
-  return useContext<BucketContext>(Context).bucket;
+export function useFlag(key: BucketFlags) {
+  const flags = useContext<BucketContext>(Context).flags;
+  const isEnabled = flags.flags[key] ?? false;
+
+  return { isLoading: flags.isLoading, isEnabled };
 }
 
 /**
  * Returns feature flags as an object, e.g.
+ * Note: this returns the raw flag keys, and does not use the
+ * optional typing provided through the `Flags` type.
  *
  * ```ts
- * const featureFlags = useFeatureFlags();
+ * const flags = useFlags();
  * // {
  * //   "isLoading": false,
  * //   "flags: {
- * //     "join-huddle": {
- * //       "key": "join-huddle",
- * //       "value": true
- * //     },
- * //     "post-message": {
- * //       "key": "post-message",
- * //       "value": true
- * //     }
+ * //     "huddle": true,
+ * //     "post-message": true
  * //   }
  * // }
  * ```
  */
-export function useFeatureFlags() {
-  const { isLoading, flags } = useContext<BucketContext>(Context).flags;
-  return { isLoading, flags };
+export function useFlags(): {
+  isLoading: boolean;
+  flags: FlagsResult;
+} {
+  const {
+    flags: { flags, isLoading },
+  } = useContext<BucketContext>(Context);
+
+  return {
+    isLoading,
+    flags,
+  };
 }
 
 /**
- * Returns the state of a given feature flag for the current context, e.g.
+ * Returns a set of functions to update the current user, company or "other context".
  *
  * ```ts
- * const joinHuddleFlag = useFeatureFlag("join-huddle");
- * // {
- * //   "isLoading": false,
- * //   "value": true,
- * // }
+ *  import { useUpdateContext } from "@bucketco/react-sdk";
+ *  function Company() {
+ *  const [company, _] = useState(initialCompany);
+ *  const { updateCompany } = useUpdateContext();
+ *  return (
+ *    <div>
+ *      <button onClick={() => updateCompany({ ...company, plan: "enterprise" })}>
+ *        Upgrade to enterprise
+ *      </button>
+ *    </div>
+ *  );
+ * }
  * ```
  */
-export function useFeatureFlag(key: string) {
-  const flags = useContext<BucketContext>(Context).flags;
-  const flag = flags.flags[key];
+export function useUpdateContext() {
+  const {
+    updateUser,
+    updateCompany,
+    updateOtherContext,
+    flags: { isLoading },
+  } = useContext<BucketContext>(Context);
+  return { updateUser, updateCompany, updateOtherContext, isLoading };
+}
```

Comment:
I would consider naming this just `useBucket` and include all methods here rather than splitting them up into their own hooks. `useFlag` hooks can stay as they alter the return in convenient ways but the others don't add much value being isolated and could simply be.

```ts
const {
  track,
  requestFeedback,
  sendFeedback,
  updateUser,
  updateCompany,
  updateOtherContext,
} = useBucket();
```

Reply:
don't you think it's either/or? weird that `useFlag` is outside `useBucket` ?

Reply:
I had a chat with @roncohen and we both agreed there were benefits and detriments to grouping it under one hook. Ultimately we decided to stick with the current approach and to test it in BETA first to get a feel for it.